src/box/lua/net_box.c

```diff
@@ -788,17 +788,27 @@ static void
 netbox_encode_select(lua_State *L, int idx, struct mpstream *stream,
 		     uint64_t sync, uint64_t stream_id)
 {
-	/* Lua stack at idx: space_id, index_id, iterator, offset, limit, key */
+	/*
+	 * Lua stack at idx: space_id, index_id, iterator, offset, limit, key,
+	 * after, fetch_pos
+	 */
 	size_t svp = netbox_begin_encode(stream, sync, IPROTO_SELECT,
 					 stream_id);
 
-	mpstream_encode_map(stream, 6);
+	uint32_t map_size = 6;
 
 	uint32_t space_id = lua_tonumber(L, idx);
 	uint32_t index_id = lua_tonumber(L, idx + 1);
 	int iterator = lua_tointeger(L, idx + 2);
 	uint32_t offset = lua_tonumber(L, idx + 3);
 	uint32_t limit = lua_tonumber(L, idx + 4);
+	bool fetch_pos = lua_toboolean(L, idx + 7);
+	if (fetch_pos)
+		map_size++;
+	if (lua_isstring(L, idx + 6) || luaT_istuple(L, idx + 6) != NULL ||
+	    lua_istable(L, idx + 6))
```

Comment:
We don't expect invalid arguments here so `!lua_isnil(L, idx_6)` would do just fine. Also, arguments are sorted by index here so `fetch_pos` should go after the position.

Reply:
Fixed